package/MDAnalysis/lib/log.py

```diff
@@ -394,3 +396,34 @@ def echo(self, step, **kwargs):
             return
         echo(self.format_handler(format, vars(self)),
              replace=self.dynamic, newline=newline)
+
+
+# detect if running from jupyter, ipython or python
+try:
+    shell = get_ipython().__class__.__name__
```

Comment:
Where does `get_ipython()` come from? Does it only exist in jupyter/ipython?

Reply:
It comes from IPython and is automatically imported when running an ipython shell or jupyter notebook/lab